mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/search/Query.java

```diff
@@ -30,6 +30,10 @@
     public static final Query ALL = new Query(ImmutableList.of());
     private static final Predicate<DeletedMessage> MATCH_ALL = any -> true;
 
+    public static Query of(List<Criterion> criteria) {
```

Comment:
it should take a single Criterion. If you want to keep things simple, you have to call this method `and` as far as I understand.

Reply:
I dont really get it, but this is the `.of()` is used, I take only criterion elements from the DTO and pass it to Query creation.
```java
class QueryTranslator {
    public Query translate(QueryDTO queryDTO) throws QueryTranslatorException {
        return Query.of(
             queryDTO.getCriteria().stream()
                   .map(queryElement -> (CriterionDTO) queryElement)
                   .map(Throwing.function(this::translate))
                   .collect(Guavate.toImmutableList()));
    }
}
```

Reply:
If you have a list, you need a combinator. `of` doesn't says if it's `and` or `or`.
If you want to make things simple, rename that method `and`.
Then you have to handle the single argument case and name this method `of`